### PR TITLE
fix. mlfow to gppi args

### PR DESCRIPTION
**Pull request by BPylypenko**
- fixes that bool parameter cannot be passed through CLI (for --tgz flag). Ref: https://stackoverflow.com/questions/15008758/parsing-boolean-values-with-argparse
or I can update to this action but it needs python3.8+
``` python
import argparse
parser = argparse.ArgumentParser()
parser.add_argument('--foo', action=argparse.BooleanOptionalAction)
parser.parse_args(['--no-foo'])
Namespace(foo=False)
```

- fixed small bug that when the `--gppi` and `--help` flags are passed, you would create folder when you passed `--gppi` flag and after that `--help` flag.
- completed revert from `--zip` to `--tgz`

### CODE REVIEW

**Comment by BPylypenko** [comment]
May I rename it from `--tgz` to `--tar` flag, if it conveys the same information?

File: mlflow/odahuflow/trainer/helpers/mlflow_helper.py [hunk 1]
```diff
@@ -258,37 +258,47 @@ def _remember_cwd():
...
+
+    parser = argparse.ArgumentParser(description='Converts MLFLow model to GPPI.')
 
     parser.add_argument('--verbose', action='store_true', help='More extensive logging')
     parser.add_argument('--model-name', type=str, required=True, help='Name of GPPI Model')
     parser.add_argument('--model-version', type=str, required=True, help='Version of GPPI Model')
-    parser.add_argument('--mlflow-model-path', '--mlflow', required=True, type=dir_type,
-                        help='Path to source MLFlow model directory')
+    parser.add_argument('--mlflow-model-path', '--mlflow', required=True,
+                        type=str, help='Path to source MLFlow model directory')
```
Comment (keshamin) [inline mlflow/odahuflow/trainer/helpers/mlflow_helper.py]: Why not using `dir_type`? Don't understand the point here.

File: mlflow/odahuflow/trainer/helpers/mlflow_helper.py [hunk 2]
```diff
@@ -258,37 +258,47 @@ def _remember_cwd():
...
     parser.add_argument('--gppi-model-path', '--gppi', required=True,
-                        type=dir_type, help='Path to result GPPI model directory')
-    parser.add_argument('--tgz', required=False, default=True,
-                        type=bool, help='Make tar arhieve with gppi folder')
-
+                        type=str, help='Path to result GPPI model directory')
+    parser.add_argument('--tgz', dest='tgz', action='store_true', help='Make tar arhieve with gppi folder')
+    parser.add_argument('--no-tgz', dest='tgz', action='store_false')
+    parser.set_defaults(tgz=True)
```
Comment (keshamin) [inline mlflow/odahuflow/trainer/helpers/mlflow_helper.py]:
```suggestion
    parser.add_argument('--no-tgz', dest='tgz', action='store_false', help='Prevent archiving result directory')
```
Using `action='store_false'` is indeed appropriate solution, but the rest is a bit overloaded in my opinion. I would suggest just to leave `--no-tgz` option. Introducing both `--tgz` and `--no-tgz` is an overhead, from my point of view.
Setting defaults is also unnecessary:
```
>>> parser = argparse.ArgumentParser()
>>> parser.add_argument('--no-bar', action='store_false', dest='bar')
>>> parser.parse_args('--no-bar'.split())
Namespace(bar=False)
>>> parser.parse_args(''.split())
Namespace(bar=True)
```

**Comment by keshamin** [comment]
> May I rename it from `--tgz` to `--tar` flag, if it conveys the same information?

As far as I know, it is not the same. We compress the tarball using gzip, so `tgz` (or `.tar.gz`) seems appropriate.

File: mlflow/odahuflow/trainer/helpers/mlflow_helper.py (hunk 2 above)
Comment (BPylypenko) [reply mlflow/odahuflow/trainer/helpers/mlflow_helper.py]: Agree. Commited

File: mlflow/odahuflow/trainer/helpers/mlflow_helper.py (hunk 1 above)
Comment (BPylypenko) [reply mlflow/odahuflow/trainer/helpers/mlflow_helper.py]: I just changed the type for `--gppi` flag and separated `dir_type` function into two (`make_dir` and `validate_dir`).
And, in the same way, I decided to change the type of `--mlflow` into `str` so it wouldn't be unique among other parameters.

But, I checked and now see that it gives a more neat error and returns errors faster.

```
: error: argument --mlflow-model-path/--mlflow: invalid dir_type value: '/Users/bohdan/wine_' 
```

So changed back to `dir_type` function.

Hence, for now, no point.)

**Review by keshamin — APPROVED** [review]

**Review by vlad-tokarev — APPROVED** [review]

**Pull request merged**